ui/src/components/core/combobox/combobox.tsx

```diff
@@ -0,0 +1,126 @@
+import { Fragment, useRef, useState } from 'react';
+import { Combobox as ComboboxLib, Transition } from '@headlessui/react';
+import { Icon, IconName } from '@/components/core/icon';
+import { Flex } from '@/components/layout';
+
+export type ComboboxItem = {
+  value: string;
+  label: string;
+  icon?: IconName;
+};
+
+export type ComboboxProps = {
+  items: ComboboxItem[];
+  selectedValue: ComboboxItem | undefined;
+  onChange(option: ComboboxItem): void;
+};
+
+export const Combobox: React.FC<ComboboxProps> = ({

```

Comment:
It would be cool to split this component in smaller ones, even if it is in the same file, just to make it simpler to understand and follow the code. For example, here we can have a component to hold the input and the search icon and also a component for options. As bigger the render get, harder is to follow and find/fix bugs.